frontend/src/services/users/service.ts

```diff
@@ -68,15 +78,63 @@ class _userService {
         firstName,
         lastName,
         email,
-        registeredAt: new Date(new Date().toISOString()),
-        updatedAt: new Date(new Date().toISOString()),
+        registeredAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
       },
     ]);
     if (error) {
       return { isError: true, message: error.message };
     }
     return { isError: false, message: "Success!" };
   };
+
+  updateAuthUser = async ({ firstName, lastName }: UpdateAuthUserRequest) => {
```

Comment:
I'm sorry but you should add validation of firstname and lastname. Because an invalidated update happens when these variables are an empty string.

```bash
curl 'http://localhost:54321/auth/v1/user' \
  -X 'PUT' \
  -H 'Accept: */*' \
  -H 'Accept-Language: en-US,en;q=0.9' \
  -H 'Authorization: Bearer ' \
  -H 'Connection: keep-alive' \
  -H 'Content-Type: application/json;charset=UTF-8' \
  -H 'Origin: http://localhost:3000' \
  -H 'Referer: http://localhost:3000/' \
  -H 'Sec-Fetch-Dest: empty' \
  -H 'Sec-Fetch-Mode: cors' \
  -H 'Sec-Fetch-Site: same-site' \
  -H 'Sec-GPC: 1' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36' \
  -H 'X-Client-Info: supabase-js/2.21.0' \
  -H 'apikey: ' \
  -H 'sec-ch-ua: "Not.A/Brand";v="8", "Chromium";v="114", "Brave";v="114"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw '{"data":{"firstName":"","lastName":""}}' \
  --compressed
```
When I run this curl command, my user name becomes blank.

![Screenshot 2023-06-30 at 12 48 31](https://github.com/e-0922-ja/expense-tracker-app/assets/12853233/4adc9dee-33d6-4d2e-8101-23369443e4a9)

Reply:
I added validations when the firstname / lastaname inputs are blank.

```js
              <StyledOutlinedInput
                {...registerUpdatedUser("firstName", {
                  required: {
                    value: true,
                    message: ERROR_EMPTY_FIRSTNAME,
                  },
                  validate: {
                    noWhitespaceOnly: (value) =>
                      value.trim() !== "" || ERROR_BLANK_FIRSTNAME,
                  },
                })}
                defaultValue={user.firstName}
                fullWidth
              />
```

Reply:
I added validation checks and some of error handlings were not working so I added try catch to handle them.
```js
// validations.ts
import {
  ERROR_BLANK_FIRSTNAME,
  ERROR_BLANK_LASTNAME,
  ERROR_EMPTY_FIRSTNAME,
  ERROR_EMPTY_LASTNAME,
} from "../../constants/message";

export const validateName = (firstName: string, lastName: string) => {
  validateFirstName(firstName);
  validateLastName(lastName);
};

export const validateFirstName = (name: string) => {
  if (!name) throw new Error(ERROR_EMPTY_FIRSTNAME);
  if (name.trim() === "") throw new Error(ERROR_BLANK_FIRSTNAME);
  return;
};

export const validateLastName = (name: string) => {
  if (!name) throw new Error(ERROR_EMPTY_LASTNAME);
  if (name.trim() === "") throw new Error(ERROR_BLANK_LASTNAME);
  return;
};
```

Reply:
Thanks!